lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -650,8 +650,8 @@ def from_ansible_params(ap, old_service):
         s.publish = []
         for param_p in ap['publish']:
             service_p = {}
-            service_p['protocol'] = param_p.get('protocol', 'tcp')
-            service_p['mode'] = param_p.get('mode', None)
+            service_p['protocol'] = param_p['protocol']
+            service_p['mode'] = param_p.get('mode')
```

Comment:
You can also simply write `param_p['mode']` here. Ansible will put in `None` if the user didn't specify it.